contracts/Vote.sol

```diff
@@ -0,0 +1,411 @@
+    }
+
+    /**
+     * @notice Retuns proposals still in referendum stage that voter voted on.
+     * @param voter The voter.
+     * @return Proposals in referendum stage.
+     * (For up to date result call updateHistoryAndReturnLockedStCeloInVoting first)
+     */
+    function getVotedStillRelevantProposals(address voter) public view returns (uint256[] memory) {
+        return voters[voter].votedProposalIds;
+    }
+
+    /**
+     * @notice Retuns currently locked celo in voting. (This celo cannot be unlocked.)
+     * @param beneficiary The account.
+     */
+    function getLockedStCeloInVoting(address beneficiary)
+        public
+        view
+        returns (uint256 lockedAmount)
+    {
+        Voter storage voter = voters[beneficiary];
+
+        uint256 i = voter.votedProposalIds.length;
+        while (i > 0) {
+            uint256 proposalId = voter.votedProposalIds[--i];
+            uint256 proposalTimestamp = proposalTimestamps[proposalId];
+
+            if (proposalTimestamp == 0) {
+                continue;
+            }
+
+            if (block.timestamp < proposalTimestamp + referendumDuration) {
+                VoterRecord storage voterRecord = voter.proposalVotes[proposalId];
+                lockedAmount = Math.max(
+                    lockedAmount,
+                    voterRecord.yesVotes + voterRecord.noVotes + voterRecord.abstainVotes
+                );
+            }
+        }
+
+        return toStakedCelo(lockedAmount);
+    }
+
+    /**
+     * @notice Sets referendum duration. It should always be the same as in Governance.
+     */
+    function setReferendumDuration() public onlyOwner {
+        uint256 newReferendumDuration = getGovernance().getReferendumStageDuration();
+        referendumDuration = newReferendumDuration;
+    }
+
+    /**
+     * @notice Returns vote weight of account owning stCelo.
+     * @param beneficiary The account.
+     */
+    function getVoteWeight(address beneficiary) public view returns (uint256) {
+        uint256 stakedCeloBalance = stakedCelo.balanceOf(beneficiary);
+        return toCelo(stakedCeloBalance);
+    }
+
+    /**
+     * @notice Gets vote record of proposal.
+     * @param proposalId The proposal UUID.
+     */
+    function getVoteRecord(uint256 proposalId) public view returns (ProposalVoteRecord memory) {
+        return voteRecords[proposalId];
+    }
+
+    /**
+     * @notice Computes the amount of stCELO that should be minted for a given
+     * amount of CELO deposited.
+     * @param celoAmount The amount of CELO deposited.
+     * @return The amount of stCELO that should be minted.
+     */
+    function toStakedCelo(uint256 celoAmount) public view returns (uint256) {
+        uint256 stCeloSupply = stakedCelo.totalSupply();
+        uint256 celoBalance = account.getTotalCelo();
+
+        if (stCeloSupply == 0 || celoBalance == 0) {
+            return celoAmount;
+        }
+
+        return (celoAmount * stCeloSupply) / celoBalance;
```

Comment:
I am a little confused what this function does, why is it multiplying the totals?